components/claim/components/ldap_to_email.tsx

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import classNames from 'classnames';
+
+import {t} from 'utils/i18n';
+import {isValidPassword, localizeMessage} from 'utils/utils';
+import {ClaimErrors} from 'utils/constants';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import {PasswordConfig} from '../claim_controller';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {SubmitOptions} from './email_to_ldap';
+
+type Props = {
+    email: string | null;
+    switchLdapToEmail: (ldapPassword: string, email: string, password: string, token: string) => Promise<{data?: AuthChangeResponse; error?: {server_error_id: string; message: string}}>;
+    passwordConfig?: PasswordConfig;
+}
+
+const LDAPToEmail = (props: Props) => {
+    const [passwordError, setPasswordError] = useState<string | JSX.Element>('');
+    const [confirmError, setConfirmError] = useState('');
+    const [ldapPasswordError, setLdapPasswordError] = useState('');
+    const [serverError, setServerError] = useState('');
+    const [showMfa, setShowMfa] = useState(true);
+    const [password, setPassword] = useState('');
+    const [ldapPassword, setLdapPassword] = useState('');
+
+    const ldapPasswordInput = useRef<HTMLInputElement>(null);
+    const passwordInput = useRef<HTMLInputElement>(null);
+    const passwordConfirmInput = useRef<HTMLInputElement>(null);
+
+    const preSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const ldapPassword = ldapPasswordInput.current?.value;
+        if (!ldapPassword) {
+            setLdapPasswordError(localizeMessage('claim.ldap_to_email.ldapPasswordError', 'Please enter your AD/LDAP password.'));
+            return;
+        }
+
+        const password = passwordInput.current?.value;
+        if (!password) {
+            setPasswordError(localizeMessage('claim.ldap_to_email.pwdError', 'Please enter your password.'));
+            return;
+        }
+
+        if (props.passwordConfig) {
+            const {valid, error} = isValidPassword(password, props.passwordConfig);
+            if (!valid && error) {
+                setPasswordError(error);
+                return;
+            }
+        }
+
+        const confirmPassword = passwordConfirmInput.current?.value;
+        if (!confirmPassword || password !== confirmPassword) {
+            setConfirmError(localizeMessage('claim.ldap_to_email.pwdNotMatch', 'Passwords do not match.'));
```

Comment:
Ditto the comment in `ldap_to_email.tsx`, we need to set all the other errors to empty in each of these cases. If we don't, we can end up showing multiple errors to the user when we only want to show the current error.

Reply:
Changes are completed thanks